core/assertion.ts

```diff
@@ -101,6 +102,7 @@ export class Assertion {
   }
 
   public query(query: AContextable<string>) {
+    throwIfTrailingSemicolonInQuery(this.session, query.toString());
```

Comment:
that `.toString()` is not going to work for anybody passing in an `AContextable` function.

this check should probably instead happen inside `.compile()`, after we have the pure, post-compilation query string.